Makefile

```diff
@@ -15,19 +15,19 @@ venv:
 	venv/bin/pip install tox
 	venv/bin/pip install -e .
 
-test:
+test: venv
```

Comment:
Should we un-PHONY the venv so that it doesn't get re-created on every `make test` etc.?

Reply:
There's also the pip install steps, but yeah those also just check that everything is already installed and then re-install cachi2

All in all it's reasonably fast

Reply:
I tried your suggestion anyway, but for some reason even if I drop venv from .PHONY make will still run the target every single time, despite the directory being present. I suspect the `--make-venv` target which doesn't have any dependency and also doesn't create a file with the same name which makes this intermediary target phony too.
So, I may have found a cleaner solution:

```
define make_venv
    $(PYTHON_BIN) -m venv $(VENV) 
    $(VENV)/bin/pip install --upgrade pip                                 
endef

...
venv:
    $(call make_venv)
...
````

By using a function instead of an intermediary target, we don't need to worry users being able to execute it. I'll update the PR accordingly.

By using a function I could also test dropping 'venv' from .PHONY. It did work, but note that the only way to re-run the 'venv' target in that case is if the directory is touched or removed, so if the contents get corrupt the target won't re-run, but I guess that's expected.